pg_lakehouse/README.md

```diff
@@ -0,0 +1,303 @@
+<h1 align="center">
+  <img src="../docs/logo/pg_lakehouse.svg" alt="pg_lakehouse" width="500px">
+<br>
+</h1>
+
+## Overview
+
+`pg_lakehouse` is an extension that transforms Postgres into an analytical query engine over object stores like S3 and table formats like Apache Iceberg. Queries are pushed down to [Apache DataFusion](https://github.com/apache/datafusion), which delivers excellent analytical performance. Combinations of the following object stores, table formats, and file formats are supported.
+
+### Object Stores
+
+- [x] Amazon S3
+- [x] S3-compatible object stores (e.g. MinIO)
+- [x] Local file system
+- [x] Google Cloud Storage
+- [ ] Azure Blob Storage (coming soon)
+
+...and potentially any service supported by [Apache OpenDAL](https://opendal.apache.org/docs/category/services). See the Development section for instructions on how to [add a service](#adding-a-service).
+
+### File Formats
+
+- [x] Parquet
+- [x] CSV
+- [x] JSON
+- [x] Avro
+- [ ] ORC (coming soon)
+
+### Table Formats
+
+- [x] Delta Lake
+- [ ] Apache Iceberg (coming soon)
+
+`pg_lakehouse` is supported on Postgres 14, 15, and 16. Support for Postgres 12 and 13 is coming soon.
+
+## Motivation
+
+Today, a vast amount of non-operational data — events, metrics, historical snapshots, vendor data, etc. — is ingested into data lakes like S3. Querying this data by moving it into a cloud data warehouse or operating a new query engine is expensive and time consuming. The goal of `pg_lakehouse` is to enable this data to be queried directly from Postgres. This eliminates the need for new infrastructure, loss of data freshness, data movement, and non-Postgres dialects of other query engines.
+
+`pg_lakehouse` uses the foreign data wrapper (FDW) API to connect to any object store or table format and the executor hook API to push queries to DataFusion. While other FDWs like `aws_s3` have existed in the Postgres extension ecosystem, these FDWs suffer from two limitations:
+
+1. Lack of support for most object stores, file, and table formats
+2. Too slow over large datasets to be a viable analytical engine
+
+`pg_lakehouse` differentiates itself by supporting a wide breadth of stores and formats (thanks to [OpenDAL](https://github.com/apache/opendal)) and by being very fast (thanks to [DataFusion](https://github.com/apache/datafusion)).
+
+## Getting Started
+
+The following example uses `pg_lakehouse` to query an example dataset of 3 million NYC taxi trips from January 2024, hosted in a public S3 bucket provided by ParadeDB.
+
+```sql
+CREATE EXTENSION pg_lakehouse;
+CREATE FOREIGN DATA WRAPPER s3_wrapper HANDLER s3_fdw_handler VALIDATOR s3_fdw_validator;
```

Comment:
I'm finding upon testing that I need to specify `paradedb.s3_fdw_handler` and `paradedb.s3_fdw_validator`, is that expected?

Reply:
I believe this was due to `paradedb` being in the schema in the `.control` file (a mistake from me, apologies). If you test with the last commit, Ming has reverted that